src/app/core/services/repo-creator.service.ts

```diff
@@ -18,9 +20,43 @@ export const BUG_REPORTING_INVALID_ROLE =
 export class RepoCreatorService {
   constructor(
     private githubService: GithubService,
-    private userService: UserService
+    private userService: UserService,
+    private phaseFixConfirmationDialog: MatDialog
   ) {}
 
+  /**
+   * Checks if the current repository has been created and if not, checks
+   * if the current phase match the requirements to make a request for
+   * repo creation permissions.
+   * @param currentPhase the current phase of the session.
+   * @param phaseRepo the name of the specified repository.
+   */
+  public requestRepoCreationPermissions(currentPhase: Phase, phaseRepo: string):
+    UnaryFunction<Observable<boolean>, Observable<boolean | null>> {
+    return pipe(
+      flatMap((isSessionAvailable: boolean) => {
```

Comment:
```suggestion
      flatMap((isRepoPresent: boolean) => {
```

I think we can drop the usage of the word `session` in this service.
It seems clearer to use `isRepoPresent` since we're focusing specifically on creating repos here.

Reply:
:+1: